<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Get started â€¢ pr2database</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Get started">
<meta property="og:description" content="pr2database">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">pr2database</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">4.12.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/pr2database.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/pr2_geo_origin.html">Tutorial - Plotting sequence location</a>
    </li>
    <li>
      <a href="../articles/pr2_stats.html">PR2 statistics</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Get started</h1>
                        <h4 class="author">Daniel Vaulot</h4>
            
      
      
      <div class="hidden name"><code>pr2database.Rmd</code></div>

    </div>

    
    
<p>The PR2 database is provided as a R package called <strong>pr2database</strong>. This page provides instruction to install and use the package.</p>
<div id="installation" class="section level1">
<h1 class="hasAnchor">
<a href="#installation" class="anchor"></a>Installation</h1>
<p>Install from the GitHub web site using the devtools package</p>
<div class="sourceCode" id="cb1"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html">install.packages</a></span>(<span class="no">devtools</span>)
<span class="kw pkg">devtools</span><span class="kw ns">::</span><span class="fu"><a href="https://devtools.r-lib.org//reference/remote-reexports.html">install_github</a></span>(<span class="st">"pr2database/pr2database"</span>)</pre></body></html></div>
<pre><code>* installing *source* package 'pr2database' ...
** R
** data
*** moving datasets to lazyload DB
** byte-compile and prepare package for lazy loading
** help
*** installing help indices
  converting help for package 'pr2database'
    finding HTML links ... fini
    pr2                                     html  
** building package indices
** testing if installed package can be loaded
*** arch - i386
*** arch - x64
* DONE (pr2database)
In R CMD INSTALL</code></pre>
</div>
<div id="loading-the-database" class="section level1">
<h1 class="hasAnchor">
<a href="#loading-the-database" class="anchor"></a>Loading the database</h1>
<p>The PR2 database is provided as a data frame (or a tibble). This is a join between the following tables: * pr2_main * pr2_taxonomy * pr2_sequence * pr2_metadata</p>
<div class="sourceCode" id="cb3"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="st">"pr2database"</span>)

<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span>(<span class="st">"pr2"</span>)

<span class="co"># List of the different columns available - see the help of the package for information on each field</span>

<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span>(<span class="no">pr2</span>)
<span class="co">#&gt;  [1] "pr2_main_id"                "pr2_accession"             </span>
<span class="co">#&gt;  [3] "genbank_accession"          "start"                     </span>
<span class="co">#&gt;  [5] "end"                        "label"                     </span>
<span class="co">#&gt;  [7] "gene"                       "organelle"                 </span>
<span class="co">#&gt;  [9] "species"                    "chimera"                   </span>
<span class="co">#&gt; [11] "chimera_remark"             "reference_sequence"        </span>
<span class="co">#&gt; [13] "added_version"              "removed_version"           </span>
<span class="co">#&gt; [15] "edited_version"             "edited_by"                 </span>
<span class="co">#&gt; [17] "edited_remark"              "remark"                    </span>
<span class="co">#&gt; [19] "taxo_id"                    "kingdom"                   </span>
<span class="co">#&gt; [21] "supergroup"                 "division"                  </span>
<span class="co">#&gt; [23] "class"                      "order"                     </span>
<span class="co">#&gt; [25] "family"                     "genus"                     </span>
<span class="co">#&gt; [27] "taxon_trophic_mode"         "taxo_edited_version"       </span>
<span class="co">#&gt; [29] "taxo_edited_by"             "taxo_removed_version"      </span>
<span class="co">#&gt; [31] "taxo_remark"                "reference"                 </span>
<span class="co">#&gt; [33] "seq_id"                     "sequence"                  </span>
<span class="co">#&gt; [35] "sequence_length"            "ambiguities"               </span>
<span class="co">#&gt; [37] "sequence_hash"              "gb_sequence"               </span>
<span class="co">#&gt; [39] "pr2_metadata_id"            "gb_date"                   </span>
<span class="co">#&gt; [41] "gb_locus"                   "gb_definition"             </span>
<span class="co">#&gt; [43] "gb_organism"                "gb_organelle"              </span>
<span class="co">#&gt; [45] "gb_taxonomy"                "gb_strain"                 </span>
<span class="co">#&gt; [47] "gb_culture_collection"      "gb_clone"                  </span>
<span class="co">#&gt; [49] "gb_isolate"                 "gb_isolation_source"       </span>
<span class="co">#&gt; [51] "gb_specimen_voucher"        "gb_host"                   </span>
<span class="co">#&gt; [53] "gb_collection_date"         "gb_environmental_sample"   </span>
<span class="co">#&gt; [55] "gb_country"                 "gb_lat_lon"                </span>
<span class="co">#&gt; [57] "gb_collected_by"            "gb_note"                   </span>
<span class="co">#&gt; [59] "gb_references"              "gb_publication"            </span>
<span class="co">#&gt; [61] "gb_authors"                 "gb_journal"                </span>
<span class="co">#&gt; [63] "pubmed_id"                  "eukref_name"               </span>
<span class="co">#&gt; [65] "eukref_source"              "eukref_env_material"       </span>
<span class="co">#&gt; [67] "eukref_env_biome"           "eukref_biotic_relationship"</span>
<span class="co">#&gt; [69] "eukref_specific_host"       "eukref_geo_loc_name"       </span>
<span class="co">#&gt; [71] "eukref_notes"               "pr2_sample_type"           </span>
<span class="co">#&gt; [73] "pr2_sample_method"          "pr2_latitude"              </span>
<span class="co">#&gt; [75] "pr2_longitude"              "pr2_ocean"                 </span>
<span class="co">#&gt; [77] "pr2_sea"                    "pr2_sea_lat"               </span>
<span class="co">#&gt; [79] "pr2_sea_lon"                "pr2_continent"             </span>
<span class="co">#&gt; [81] "pr2_country"                "pr2_location"              </span>
<span class="co">#&gt; [83] "pr2_location_geoname"       "pr2_location_geotype"      </span>
<span class="co">#&gt; [85] "pr2_location_lat"           "pr2_location_lon"          </span>
<span class="co">#&gt; [87] "pr2_country_geocode"        "pr2_country_lat"           </span>
<span class="co">#&gt; [89] "pr2_country_lon"            "pr2_sequence_origin"       </span>
<span class="co">#&gt; [91] "pr2_size_fraction"          "pr2_size_fraction_min"     </span>
<span class="co">#&gt; [93] "pr2_size_fraction_max"      "metadata_remark"</span></pre></body></html></div>
</div>
<div id="working-with-the-database" class="section level1">
<h1 class="hasAnchor">
<a href="#working-with-the-database" class="anchor"></a>Working with the database</h1>
<div id="install-and-load-the-libraries" class="section level2">
<h2 class="hasAnchor">
<a href="#install-and-load-the-libraries" class="anchor"></a>Install and load the libraries</h2>
<p>The following examples makes use of the specifc R libraries</p>
<p>Install the libraries</p>
<div class="sourceCode" id="cb4"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html">install.packages</a></span>(<span class="st">"dplyr"</span>)      <span class="co"># For filtering the data</span>
<span class="fu">install.package</span>(<span class="st">"ggplot2"</span>)     <span class="co"># To plot data</span>
<span class="fu">install.package</span>(<span class="st">"maps"</span>)        <span class="co"># To plot maps</span>

<span class="fu"><a href="https://rdrr.io/r/base/source.html">source</a></span>(<span class="st">"https://bioconductor.org/biocLite.R"</span>)  <span class="co"># This package is on Bioconductor</span>
<span class="fu">biocLite</span>(<span class="st">"Biostrings"</span>)         <span class="co"># To save fasta files</span></pre></body></html></div>
<p>Load the libraries</p>
<div class="sourceCode" id="cb5"><html><body><pre class="r">  <span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">dplyr</span>)
  <span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">ggplot2</span>)    <span class="co"># For plots</span>
  <span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">Biostrings</span>) <span class="co"># To save fasta files</span></pre></body></html></div>
</div>
<div id="selecting-sequences-from-a-specific-taxon" class="section level2">
<h2 class="hasAnchor">
<a href="#selecting-sequences-from-a-specific-taxon" class="anchor"></a>Selecting sequences from a specific taxon</h2>
<p>Let us select all the available sequences for the Mamiellophyceae <em>Ostreococcus</em></p>
<div class="sourceCode" id="cb6"><html><body><pre class="r">
  <span class="co"># Filter only the sequences for which the column genus contains Ostreococcus</span>
  <span class="no">pr2_ostreo</span> <span class="kw">&lt;-</span> <span class="no">pr2</span> <span class="kw">%&gt;%</span> <span class="kw pkg">dplyr</span><span class="kw ns">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span>(<span class="no">genus</span> <span class="kw">==</span> <span class="st">"Ostreococcus"</span>)

  <span class="co"># Select only the columns of interest</span>
  <span class="no">pr2_ostreo</span> <span class="kw">&lt;-</span> <span class="no">pr2_ostreo</span> <span class="kw">%&gt;%</span> <span class="kw pkg">dplyr</span><span class="kw ns">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html">select</a></span>( <span class="no">genbank_accession</span>, <span class="no">species</span>,
                                              <span class="no">pr2_sample_type</span>, <span class="no">gb_strain</span>, <span class="no">gb_clone</span>,
                                              <span class="no">pr2_latitude</span>, <span class="no">pr2_longitude</span>,
                                              <span class="no">sequence_length</span>, <span class="no">sequence</span>  )

  <span class="no">pr2_ostreo</span>
<span class="co">#&gt; # A tibble: 325 x 9</span>
<span class="co">#&gt;    genbank_accessi~ species pr2_sample_type gb_strain gb_clone pr2_latitude</span>
<span class="co">#&gt;    &lt;chr&gt;            &lt;chr&gt;   &lt;chr&gt;           &lt;chr&gt;     &lt;chr&gt;           &lt;dbl&gt;</span>
<span class="co">#&gt;  1 AF525872         Ostreo~ environmental   &lt;NA&gt;      UEPACIp5         NA  </span>
<span class="co">#&gt;  2 EU562149         Ostreo~ environmental   &lt;NA&gt;      IND2.6           NA  </span>
<span class="co">#&gt;  3 AY425309         Ostreo~ environmental   &lt;NA&gt;      RA01041~         NA  </span>
<span class="co">#&gt;  4 GQ426346         Ostreo~ culture         CB6       &lt;NA&gt;             NA  </span>
<span class="co">#&gt;  5 KC583118         Ostreo~ environmental   &lt;NA&gt;      RS.12f.~         NA  </span>
<span class="co">#&gt;  6 JN862906         Ostreo~ culture         BCC48000  &lt;NA&gt;             NA  </span>
<span class="co">#&gt;  7 JQ692065         Ostreo~ environmental   &lt;NA&gt;      PUPF_60         -43.3</span>
<span class="co">#&gt;  8 FR874749         Ostreo~ environmental   &lt;NA&gt;      1815F12          60.3</span>
<span class="co">#&gt;  9 FJ431431         Ostreo~ environmental   &lt;NA&gt;      RA07100~         NA  </span>
<span class="co">#&gt; 10 EU561670         Ostreo~ environmental   &lt;NA&gt;      IND1.11         -35.0</span>
<span class="co">#&gt; # ... with 315 more rows, and 3 more variables: pr2_longitude &lt;dbl&gt;,</span>
<span class="co">#&gt; #   sequence_length &lt;int&gt;, sequence &lt;chr&gt;</span></pre></body></html></div>
</div>
<div id="exporting-the-sequences-to-fasta" class="section level2">
<h2 class="hasAnchor">
<a href="#exporting-the-sequences-to-fasta" class="anchor"></a>Exporting the sequences to fasta</h2>
<p>We will save the <em>Ostreococcus</em> sequences to a FASTA file. This is easy done with the bioconductor package BioStrings.</p>
<div class="sourceCode" id="cb7"><html><body><pre class="r">
  <span class="co"># Importing the sequence in a Biostring set </span>

  <span class="no">seq_ostreo</span> <span class="kw">&lt;-</span> <span class="kw pkg">Biostrings</span><span class="kw ns">::</span><span class="fu"><a href="https://rdrr.io/pkg/Biostrings/man/XStringSet-class.html">DNAStringSet</a></span>(<span class="no">pr2_ostreo</span>$<span class="no">sequence</span>)

  <span class="co"># Constructing the name of each sequecne (the first line of the fasta file)</span>
  <span class="co"># using the genbank accession, species name, strain name and clone name</span>

  <span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span>(<span class="no">seq_ostreo</span>) <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste</a></span>(<span class="no">pr2_ostreo</span>$<span class="no">genbank_accession</span>, <span class="no">pr2_ostreo</span>$<span class="no">species</span>,
                             <span class="st">"strain"</span>,<span class="no">pr2_ostreo</span>$<span class="no">gb_strain</span>,
                             <span class="st">"clone"</span>,<span class="no">pr2_ostreo</span>$<span class="no">gb_clone</span>,
                              <span class="kw">sep</span><span class="kw">=</span><span class="st">"|"</span>)

  <span class="co"># Displaying the Biostring set</span>
  <span class="no">seq_ostreo</span>
<span class="co">#&gt;   A DNAStringSet instance of length 325</span>
<span class="co">#&gt;       width seq                                             names               </span>
<span class="co">#&gt;   [1]  1766 ACCTGGTTGATCCTGCCAGTAG...AGGTGAACCTGCAGAAGGATCA AF525872|Ostreoco...</span>
<span class="co">#&gt;   [2]   836 AAAGCTCGTAGTCGGATTTTGG...TCTGGGCCGCACGCGCGCTACA EU562149|Ostreoco...</span>
<span class="co">#&gt;   [3]  1728 GCCAGTAGTCATATGCTTGTCT...GAGAAGTCGTAACAAGGTTTCC AY425309|Ostreoco...</span>
<span class="co">#&gt;   [4]  1652 AGCCATGCATGTCTAAGTATAA...TGGATTACCGTGGGAAATTCGT GQ426346|Ostreoco...</span>
<span class="co">#&gt;   [5]  1764 CCTGGTTGATCCTGCCAGTAGT...TAGGTGAACCTGCAGAAGGATC KC583118|Ostreoco...</span>
<span class="co">#&gt;   ...   ... ...</span>
<span class="co">#&gt; [321]  1570 CAATTTGAATGAGATTCAAAAA...TCAAAAAAGACCAAGCCGGAAG KF285522|Ostreoco...</span>
<span class="co">#&gt; [322]  1570 CAATTTGAATGAGATTCAAAAA...TCAAAAAAGACCAAGCCGGAAG KF285529|Ostreoco...</span>
<span class="co">#&gt; [323]  1570 CAATTTGAATGAGATTCAAAAA...TCAAAAAAGACCAAGCCGGAAG KF285531|Ostreoco...</span>
<span class="co">#&gt; [324]  1570 CAATTTGAATGAGATTCAAAAA...TCAAAAAAGACCAAGCCGGAAG NC_008289|Ostreoc...</span>
<span class="co">#&gt; [325]  1570 CAATTTGAATGAGATTCAAAAA...TCAAAAAAGACCAAGCCGGAAG NC_008289|Ostreoc...</span>

  <span class="co"># Saving the sequences as a fasta file</span>
  <span class="kw pkg">Biostrings</span><span class="kw ns">::</span><span class="fu"><a href="https://rdrr.io/pkg/Biostrings/man/XStringSet-io.html">writeXStringSet</a></span>(<span class="no">seq_ostreo</span>, <span class="st">"examples/pr2_ostreo.fasta"</span>, <span class="kw">width</span> <span class="kw">=</span> <span class="fl">80</span>)</pre></body></html></div>
<p>The fasta file will look as follows</p>
<pre><code>&gt;AF525872|Ostreococcus_lucimarinus|strain|NA|clone|UEPACIp5
ACCTGGTTGATCCTGCCAGTAGTCATATGCTTGTCTCAAAGATTAAGCCATGCATGTCTAAGTATAAGCGTTATACTGTG
AAACTGCGAATGGCTCATTAAATCAGCAATAGTTTCTTTGGTGGTGTTTACTACTCGGATAACCGTAGTAATTCTAGAGC
TAATACGTGCGTAAATCCCGACTTCGGAAGGGACGTATTTATTAGATAAAGACCG...
&gt;EU562149|Ostreococcus_lucimarinus|strain|NA|clone|IND2.6
AAAGCTCGTAGTCGGATTTTGGCTGAGAACGGTCGGTCCGCCGTTAGGTGTGCACTGACTGGTCTCAGCTTCCTGGTGAG
GAGGTGTGCTTCATCGCCACTTAGTCACCGTGGTTACTTTGAAAAAATTAGAGTGTTCAAAGCGGGCTTACGCTTGAATA
TATTAGCATGGAATAACACCATAGGACTCCTGTCCTATTTCGTTGGTCTCGGGACGGGAGTAATGATTAAGATGAACAGT
TGGGGGCATTCGTATTTCATTGTCAGAGGTGAAATTCTTGGATTT...
&gt;AY425309|Ostreococcus_lucimarinus|strain|NA|clone|RA010412.39
GCCAGTAGTCATATGCTTGTCTCAAAGATTAAGCCATGCATGTCTAAGTATAAGCGTTATACTGTGAAACTGCGAATGGC
TCATTAAATCAGCAATAGTTTCTTTGGTGGTGTTTACTACTCGGATAACCGT...</code></pre>
</div>
<div id="doing-an-histogram-of-the-sequence-length" class="section level2">
<h2 class="hasAnchor">
<a href="#doing-an-histogram-of-the-sequence-length" class="anchor"></a>Doing an histogram of the sequence length</h2>
<div class="sourceCode" id="cb9"><html><body><pre class="r">  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html">ggplot</a></span>(<span class="no">pr2_ostreo</span>) +
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_histogram.html">geom_histogram</a></span>(<span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span>(<span class="no">sequence_length</span>), <span class="kw">binwidth</span> <span class="kw">=</span> <span class="fl">50</span>, <span class="kw">fill</span><span class="kw">=</span><span class="st">"blue"</span>) +
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/lims.html">xlim</a></span>(<span class="fl">0</span>,<span class="fl">2000</span>) + <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">xlab</a></span>(<span class="st">"Sequence length"</span>) + <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">ylab</a></span>(<span class="st">"Number of sequences"</span>) +
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">ggtitle</a></span>(<span class="st">"Ostreococcus lucimarinus"</span>)</pre></body></html></div>
<p><img src="pr2database_files/figure-html/sequence_histogram-1.png" width="700"></p>
</div>
<div id="drawing-a-map-of-sequence-locations" class="section level2">
<h2 class="hasAnchor">
<a href="#drawing-a-map-of-sequence-locations" class="anchor"></a>Drawing a map of sequence locations</h2>
<div class="sourceCode" id="cb10"><html><body><pre class="r">  <span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">maps</span>)
  <span class="no">world</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/map_data.html">map_data</a></span>(<span class="st">"world"</span>)

  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html">ggplot</a></span>() +
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_polygon.html">geom_polygon</a></span>(<span class="kw">data</span> <span class="kw">=</span> <span class="no">world</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span>(<span class="kw">x</span><span class="kw">=</span><span class="no">long</span>, <span class="kw">y</span> <span class="kw">=</span> <span class="no">lat</span>, <span class="kw">group</span> <span class="kw">=</span> <span class="no">group</span>), <span class="kw">fill</span><span class="kw">=</span><span class="st">"grey"</span>) +
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/coord_fixed.html">coord_fixed</a></span>(<span class="fl">1.3</span>) +
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html">geom_point</a></span>(<span class="kw">data</span><span class="kw">=</span><span class="no">pr2_ostreo</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span>(<span class="kw">x</span><span class="kw">=</span><span class="no">pr2_longitude</span>, <span class="kw">y</span><span class="kw">=</span><span class="no">pr2_latitude</span>), <span class="kw">fill</span><span class="kw">=</span><span class="st">"blue"</span>, <span class="kw">size</span><span class="kw">=</span><span class="fl">2</span>, <span class="kw">shape</span><span class="kw">=</span><span class="fl">21</span>) +
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">ggtitle</a></span>(<span class="st">"Ostreococcus lucimarinus"</span>)</pre></body></html></div>
<p><img src="pr2database_files/figure-html/sequence_map-1.png" width="700"></p>
</div>
<div id="drawing-a-map-of-sequence-locations-obtained-by-fuzzy-matching" class="section level2">
<h2 class="hasAnchor">
<a href="#drawing-a-map-of-sequence-locations-obtained-by-fuzzy-matching" class="anchor"></a>Drawing a map of sequence locations obtained by fuzzy matching</h2>
<p>A very good tutorial by <a href="https://twitter.com/MargaretBrisbin">Margaret Mars Brisbin</a> on how to combine PR2 metadata with Python to locate sequences using all the metadata information (lat, long, country and fuzzy localization): <a href="https://maggimars.github.io/eukGeoBlast/eGB.html" class="uri">https://maggimars.github.io/eukGeoBlast/eGB.html</a> and <a href="https://github.com/maggimars/eukGeoBlast" class="uri">https://github.com/maggimars/eukGeoBlast</a>. This code has been used to incorporate more geo-localisation information into PR2 version 4.12.0</p>
<p><img src="https://github.com/pr2database/pr2database/blob/master/img/pr2_loc_fuzzy.png"></p>
</div>
<div id="number-of-sequences-per-country" class="section level2">
<h2 class="hasAnchor">
<a href="#number-of-sequences-per-country" class="anchor"></a>Number of sequences per country</h2>
<p>Version 4.12.0 incorporate better geo-localisation using the approach pioneered by<a href="https://twitter.com/MargaretBrisbin">Margaret Mars Brisbin</a></p>
<div id="number-of-sequences-per-country-of-origin" class="section level3">
<h3 class="hasAnchor">
<a href="#number-of-sequences-per-country-of-origin" class="anchor"></a>Number of sequences per country of origin</h3>
<div class="sourceCode" id="cb11"><html><body><pre class="r">
  <span class="no">countries</span> <span class="kw">&lt;-</span> <span class="no">pr2</span> <span class="kw">%&gt;%</span>
    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/count.html">count</a></span>(<span class="no">pr2_country</span>) <span class="kw">%&gt;%</span>
    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/arrange.html">arrange</a></span>(-<span class="no">n</span>) <span class="kw">%&gt;%</span>
    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span>(!<span class="fu"><a href="https://rdrr.io/r/base/NA.html">is.na</a></span>(<span class="no">pr2_country</span>) <span class="kw">&amp;</span> <span class="no">n</span> <span class="kw">&gt;</span> <span class="fl">500</span>)

  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html">ggplot</a></span>(<span class="no">countries</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span>(<span class="kw">x</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/stats/reorder.factor.html">reorder</a></span>(<span class="no">pr2_country</span>, <span class="no">n</span>), <span class="kw">y</span> <span class="kw">=</span> <span class="no">n</span>)) +
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_bar.html">geom_col</a></span>() +
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/coord_flip.html">coord_flip</a></span>() +
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">xlab</a></span>(<span class="st">""</span>)  + <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">ylab</a></span>(<span class="st">"Number of PR2 sequences"</span>)</pre></body></html></div>
<p><img src="pr2database_files/figure-html/sequence_country-1.png" width="700"></p>
</div>
<div id="number-of-sequences-per-ocean-of-origin" class="section level3">
<h3 class="hasAnchor">
<a href="#number-of-sequences-per-ocean-of-origin" class="anchor"></a>Number of sequences per ocean of origin</h3>
<div class="sourceCode" id="cb12"><html><body><pre class="r">
  <span class="no">oceans</span> <span class="kw">&lt;-</span> <span class="no">pr2</span> <span class="kw">%&gt;%</span>
    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/count.html">count</a></span>(<span class="no">pr2_ocean</span>) <span class="kw">%&gt;%</span>
    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/arrange.html">arrange</a></span>(-<span class="no">n</span>) <span class="kw">%&gt;%</span>
    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span>(!<span class="fu"><a href="https://rdrr.io/r/base/NA.html">is.na</a></span>(<span class="no">pr2_ocean</span>))

  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html">ggplot</a></span>(<span class="no">oceans</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span>(<span class="kw">x</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/stats/reorder.factor.html">reorder</a></span>(<span class="no">pr2_ocean</span>, <span class="no">n</span>), <span class="kw">y</span> <span class="kw">=</span> <span class="no">n</span>)) +
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_bar.html">geom_col</a></span>() +
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/coord_flip.html">coord_flip</a></span>() +
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">xlab</a></span>(<span class="st">""</span>)  + <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">ylab</a></span>(<span class="st">"Number of PR2 sequences"</span>)</pre></body></html></div>
<p><img src="pr2database_files/figure-html/sequence_oceans-1.png" width="700"></p>
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Daniel Vaulot.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.5.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
