<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>PR2 database of 18S rRNA sequences. — pr2 • pr2database</title>


<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous" />

<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="PR2 database of 18S rRNA sequences. — pr2" />
<meta property="og:description" content="The PR2 database is provided as a data frame.  This is a join between the following tables:
pr2_main
pr2_taxonomy
pr2_sequence
pr2_metadata
The metadata contains three types of fields
gb_ : originating from the GenBank entry
eukref_ : annotated by the Eukref project
pr2_ : annotated by pr2 such latitude and longitude

" />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">pr2database</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">4.12.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/pr2database.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/pr2_geo_origin.html">Tutorial - Plotting sequence location</a>
    </li>
    <li>
      <a href="../articles/pr2_stats.html">PR2 statistics</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>PR2 database of 18S rRNA sequences.</h1>
    
    <div class="hidden name"><code>pr2.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>The PR2 database is provided as a data frame.  This is a join between the following tables:</p><ul>
<li><p>pr2_main</p></li>
<li><p>pr2_taxonomy</p></li>
<li><p>pr2_sequence</p></li>
<li><p>pr2_metadata</p></li>
</ul><p>The metadata contains three types of fields</p><ul>
<li><p>gb_ : originating from the GenBank entry</p></li>
<li><p>eukref_ : annotated by the Eukref project</p></li>
<li><p>pr2_ : annotated by pr2 such latitude and longitude</p></li>
</ul>

    </div>

    <pre class="usage"><span class='no'>pr2</span></pre>


    <h2 class="hasAnchor" id="format"><a class="anchor" href="#format"></a>Format</h2>

    <p>A data frame : 183,919 x 93:</p><dl'>
  <dt>pr2_main_id</dt><dd><p>ID of PR2 entry in table pr2_main</p></dd>
  <dt>pr2_accession</dt><dd><p>PR2 specific accession number</p></dd>
  <dt>genbank_accession</dt><dd><p>Genbank accession number (without the vresion)</p></dd>
  <dt>start</dt><dd><p>Start of sequence in Genbank entry</p></dd>
  <dt>end</dt><dd><p>End of sequence in Genbank entry</p></dd>
  <dt>label</dt><dd><p>Label explaining origin of sequence</p></dd>
  
       G: genomic sequence containing a described intron (rDNA)
       R: the previous genomic rRNA sequence, without the intron(s)
       U: no intron described, but intron(s) may be present
       UC: introns were detected in silico and removed from the sequence (putative rRNA) 
  
  <dt>gene</dt><dd><p>18S_rRNA or 16S_rRNA</p></dd>
  <dt>organelle</dt><dd><p>nucleus, plastid, mitochondria, nucleomorph, apicoplast (left empty for cyanobacteria)</p></dd>
  <dt>species</dt><dd><p>Assigned species - rank 8</p></dd>
  <dt>chimera</dt><dd><p>= 1, the sequence is chimeric</p></dd>
  <dt>chimera_remark</dt><dd><p>Comment on reason why the sequence was mentionned as chimeric</p></dd>
  <dt>reference_sequence</dt><dd><p>= 1, this is a reference sequence that can be used for example for alignements</p></dd>
  <dt>added_version</dt><dd><p>PR2 version when sequence was added</p></dd>
  <dt>removed_version</dt><dd><p>PR2 version when sequence was removed</p></dd>
  <dt>edited_version</dt><dd><p>PR2 version when sequence assignation was edited</p></dd>
  <dt>edited_by</dt><dd><p>Who edited the assignation</p></dd>
  <dt>edited_remark</dt><dd><p>Remark concerning the assignation</p></dd>
  <dt>remark</dt><dd><p>Remark concerning  the sequence</p></dd>
  <dt>taxo_id</dt><dd><p>ID of the species in table pr2_taxonomy</p></dd>
  <dt>kingdom</dt><dd><p>rank 1</p></dd>
  <dt>supergroup</dt><dd><p>rank 2</p></dd>
  <dt>division</dt><dd><p>rank 3</p></dd>
  <dt>class</dt><dd><p>rank 4</p></dd>
  <dt>order</dt><dd><p>rank 5</p></dd>
  <dt>family</dt><dd><p>rank 6</p></dd>
  <dt>genus</dt><dd><p>rank 7</p></dd>
  <dt>taxon_trophic_mode</dt><dd><p>e.g. symbiotic</p></dd>
  <dt>taxo_edited_version</dt><dd><p>PR2 version when taxonomy path was edited</p></dd>
  <dt>taxo_edited_by</dt><dd><p>Who edited the taxonomy path</p></dd>
  <dt>taxo_removed_version</dt><dd><p>PR2 version when species and taxonomy path was removed</p></dd>
  <dt>taxo_remark</dt><dd><p>Remark concerning  the taxonomy</p></dd>
  <dt>reference</dt><dd><p>Reference in the litterature concerning  the taxonomy</p></dd>
  <dt>seq_id</dt><dd><p>ID of the sequence in table pr2_sequence</p></dd>
  <dt>sequence</dt><dd><p>Sequence</p></dd>
  <dt>sequence_length</dt><dd><p>Length of sequence</p></dd>
  <dt>ambiguities</dt><dd><p>Number of ambiguities</p></dd>
  <dt>sequence_hash</dt><dd><p>Hash value of sequence</p></dd>
  <dt>pr2_metadata_id</dt><dd><p>ID of metadata in table pr2_metadata</p></dd>
  <dt>gb_date</dt><dd><p>Genbank: Date</p></dd>
  <dt>gb_locus</dt><dd><p>Genbank: Locus</p></dd>
  <dt>gb_definition</dt><dd><p>Genbank: Definition</p></dd>
  <dt>gb_organism</dt><dd><p>Genbank: Organism</p></dd>
  <dt>gb_taxonomy</dt><dd><p>Genbank: Taxonomy</p></dd>
  <dt>gb_strain</dt><dd><p>Genbank: Strain</p></dd>
  <dt>gb_culture_collection</dt><dd><p>Genbank: Culture Collection</p></dd>
  <dt>gb_clone</dt><dd><p>Genbank: Clone</p></dd>
  <dt>gb_isolate</dt><dd><p>Genbank: Isolate</p></dd>
  <dt>gb_isolation_source</dt><dd><p>Genbank: Isolation Source</p></dd>
  <dt>gb_specimen_voucher</dt><dd><p>Genbank: Voucher</p></dd>
  <dt>gb_host</dt><dd><p>Genbank: Host</p></dd>
  <dt>gb_collection_date</dt><dd><p>Genbank: Date of Collection</p></dd>
  <dt>gb_environmental_sample</dt><dd><p>Genbank: Environmental Sample</p></dd>
  <dt>gb_country</dt><dd><p>Genbank: Country</p></dd>
  <dt>gb_lat_lon</dt><dd><p>Genbank: lat Lon</p></dd>
  <dt>gb_collected_by</dt><dd><p>Genbank: Collected by</p></dd>
  <dt>gb_note</dt><dd><p>Genbank: Note</p></dd>
  <dt>gb_references</dt><dd><p>Genbank: Full references not parsed</p></dd>
  <dt>gb_publication</dt><dd><p>Genbank: Publication</p></dd>
  <dt>gb_authors</dt><dd><p>Genbank: Authors</p></dd>
  <dt>gb_journal</dt><dd><p>Genbank: Journal</p></dd>
  <dt>pubmed_id</dt><dd><p>Genbank: Pubmed ID</p></dd>
  <dt>eukref_name</dt><dd><p>Eukref: Name use in EukRef, usually either the species name or the clone name</p></dd>
  <dt>eukref_source</dt><dd><p>Eukref: Source of the sequence : Isolate or Environmental</p></dd>
  <dt>eukref_env_material</dt><dd><p>Eukref: uses ENVO keywords</p></dd>
  <dt>eukref_env_biome</dt><dd><p>Eukref: uses ENVO keywords</p></dd>
  <dt>eukref_biotic_relationship</dt><dd><p>Eukref: eg parasite</p></dd>
  <dt>eukref_specific_host</dt><dd><p>Eukref: Specific Host annotated</p></dd>
  <dt>eukref_geo_loc_name</dt><dd><p>Eukref: Location name annotated</p></dd>
  <dt>eukref_notes</dt><dd><p>Eukref: Notes made during Eukref annotation</p></dd>
  <dt>pr2_sample_type</dt><dd><p>PR2: e.g. water, ice</p></dd>
  <dt>pr2_sample_method</dt><dd><p>PR2: e.g. filtration, flow cytometry sorting</p></dd>
  <dt>pr2_latitude</dt><dd><p>PR2: Parsed from GenBank entry</p></dd>
  <dt>pr2_longitude</dt><dd><p>PR2: Parsed from GenBank entry</p></dd>
  <dt>pr2_ocean</dt><dd><p>PR2: e.g. Arctic Ocean</p></dd>
  <dt>pr2_sea</dt><dd><p>PR2: e.g. North Sea</p></dd>
  <dt>pr2_sea_lat</dt><dd><p>PR2: latitude of sea or ocean</p></dd>
  <dt>pr2_sea_lon</dt><dd><p>PR2: longitude of sea or ocean</p></dd>
  <dt>pr2_continent</dt><dd><p>PR2: e.g. Asia</p></dd>
  <dt>pr2_country</dt><dd><p>PR2: e.g. France</p></dd>
  <dt>pr2_country_geocode</dt><dd><p>PR2: 2 letter code from genonames - e.g. FR</p></dd>
  <dt>pr2_country_lat</dt><dd><p>PR2: latitude of country</p></dd>
  <dt>pr2_country_lon</dt><dd><p>PR2: longitude of country</p></dd>
  <dt>pr2_location</dt><dd><p>PR2: from gb_country field - e.g. Paris, France</p></dd>
  <dt>pr2_location_geoname</dt><dd><p>PR2: e.g. Paris</p></dd>
  <dt>pr2_location_geotype</dt><dd><p>PR2: e.g. bay</p></dd>
  <dt>pr2_location_lat</dt><dd><p>PR2: latitude of location</p></dd>
  <dt>pr2_location_lon</dt><dd><p>PR2: longtitude of location</p></dd>
  <dt>pr2_sequence_origin</dt><dd><p>PR2: clone library, metabarcode, PCR</p></dd>
  <dt>pr2_size_fraction</dt><dd><p>PR2: Name of size fraction, e.g. pico, nano</p></dd>
  <dt>pr2_size_fraction_min</dt><dd><p>PR2: Minimum size filtered, e.g. 0.2 µm</p></dd>
  <dt>pr2_size_fraction_max</dt><dd><p>PR2: Maximum size filtered, e.g. 20 µm</p></dd>
  <dt>metadata_remark</dt><dd><p>PR2: Any remark on metadata</p></dd>

</dl>
    <h2 class="hasAnchor" id="source"><a class="anchor" href="#source"></a>Source</h2>

    <p>PR2 database <a href='https://github.com/vaulot/pr2database'>https://github.com/vaulot/pr2database</a></p>

  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by Daniel Vaulot.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.5.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


